analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ExplodedNode.cs

```diff
@@ -46,8 +46,11 @@ private ExplodedNode(BasicBlock block, IOperationWrapperSonar[] operations, int
             programPointHash = ProgramPoint.Hash(block, index);
         }
 
-        public ExplodedNode CreateNext(ProgramState state) =>
-            new(Block, operations, index + 1, state, FinallyPoint);
+        public ExplodedNode CreateNext(ProgramState state)
+        {
+            state?.CheckStateConsistency();
```

Comment:
The "state" reads as duplicate
```suggestion
            state?.CheckConsistency();
```

Reply:
What about something like this? 
```
new(Block, operations, index + 1, state.EnsureConsistency(), FinallyPoint);
```
Where the implementation will do
```
CheckConsistency(); // <--- this is conditional
return this;
```